core/dbt/parser/sources.py

```diff
@@ -154,7 +155,48 @@ def parse_source(self, target: UnpatchedSourceDefinition) -> SourceDefinition:
                 loaded_at_query = None
             else:
                 loaded_at_query = source.loaded_at_query
-        freshness = merge_freshness(source.freshness, table.freshness)
+
+        try:
+            project_freshness = FreshnessThreshold.from_dict(
+                self.root_project.sources.get("freshness", {})
+            )
+        except ValueError:
+            fire_event(
+                FreshnessConfigProblem(
+                    msg="Could not parse default freshness config in 'dbt_project.yml', ignoring.",
+                )
+            )
+            project_freshness = None
+
+        source_freshness = source.freshness
+        if source_freshness:
+            deprecations.warn(
+                "custom-key-in-object-deprecation",
+                key="freshness",
+                file=target.path,
+                key_path=source.name,
```

Comment:
Can we create a more specific deprecation for this? My concern is that the `custom-key-in-object-deprecation` basically indicates that we don't know what the property is. It might be more useful to the user if we create something like a `freshness-property-moved-deprecation`. Where we give a message like.
```
[WARNING]: Deprecated functionality
Found `freshness` as a top level property of `<resource_name>` in file `<relative_file_path>`. The `freshness` specification should be moved into the `config` of the resource.
```
The idea being, we want to make it clear to the user that defining `freshness` isn't deprecated, but where one specifies it has been.